### ISSUE TITLE
Parameter tuning when training with a NVIDIA 4090 GPU

### ISSUE THREAD

**Issue by Yuan-Xinyi** [tylerlum/simtoolreal#21]
Thanks for the great work and for making all the code open source. I am trying to train the model with a 4090; however, the default number of environments exceeds the 4090's GPU memory. I am trying to train the model on only half of the environment number, but the success rate didn't improve much and still stays around 0.

I was wondering if there are other parameters that need to be tuned along with the environment number. Are there recommended hyperparameter changes (e.g., learning rate, batch size, rollout length) to compensate for the reduced number of envs?

**Comment by kushal2000** [comment]
Hi Xinyi, we have tested before with 12k environments and the training run works for us. In this training command, I would overwrite two parameters, `num_envs` and `expl_coef_block_size`. Dividing both by half keep the total number of blocks used in SAPG constant.

```
.venv_isaacsim/bin/python isaacsimenvs/train.py \
--task Isaacsimenvs-SimToolReal-Direct-v0 \
--agent rl_games_sapg_cfg_entry_point \
--headless \
--capture_viewer \
--wandb_activate \
--wandb_project <project> \
--wandb_entity <entity> \
env.scene.num_envs=12288 \
agent.params.config.expl_coef_block_size=2046
```

With lesser number of envs, you might need more training time (usually you should see signs of life within 2 days)

**Comment by Yuan-Xinyi** [comment]
Thank you so much for the detailed guidance and for sharing your setup. This is really helpful. 

I'll go ahead and adjust num_envs and expl_coef_block_size as you suggested, and wait for the signs of life over the next couple of days. I'd be happy to share any progress later. Thanks again for your help!

**Comment by Abao2** [comment]
Hi, thanks for sharing your experience here. I'm currently trying to reproduce the training as well, so I was wondering if you could share a quick update on how it went after adjusting num_envs and expl_coef_block_size.

Did you eventually see the expected “signs of life” and manage to reproduce reasonable performance? If so, roughly how long did it take, and were there any other changes to the default configuration that turned out to be important?

Thanks a lot!

**Comment by Yuan-Xinyi** [comment]
Hi Abao, sorry, I used A6000 afterward, so I directly applied the original parameter. The live sign appeared after at least 1 day. I guess if you use the half parameter, you need to wait for 2-3 days.
